analyzers/tests/SonarAnalyzer.Test/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -172,6 +173,56 @@ public void ArrowExpressionBody_WithNotExpectedNode_ReturnsNull()
         public void GetDeclarationTypeName_Struct() =>
             ExtensionsCS.GetDeclarationTypeName(SyntaxFactory.StructDeclaration("MyStruct")).Should().Be("struct");
 
+        [TestMethod]
+        public void GetBody_FieldDeclaration()
+        {
+            ExtensionsCS.GetBody(SyntaxFactory.FieldDeclaration(SyntaxFactory.VariableDeclaration(SyntaxFactory.ParseTypeName("int")))).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_PropertyDeclaration_SingleAccessor()
+        {
+            var code = "class AClass { int AProperty { get => 42; } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<PropertyDeclarationSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_PropertyDeclaration_MultipleAccessors()
+        {
+            var code = "class AClass { int AProperty { get => 42; set { } } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<PropertyDeclarationSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_AccessorDeclaration_ArrowExpression()
+        {
+            var code = "class AClass { int AProperty { get => 42; } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<AccessorDeclarationSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_AccessorDeclaration_BodyBlock()
+        {
+            var code = "class AClass { int AProperty { set { var x1; var x2; var x3; } } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<AccessorDeclarationSyntax>())
+                .Should().BeOfType<BlockSyntax>().Which.Statements.Should().HaveCount(3);
+        }
+
+        [TestMethod]
+        public void GetBody_LocalFunctionStatement_ArrowExpression()
+        {
+            var code = "class AClass { void AMethod() { int ALocalFunction() => 42; } }";
+            ExtensionsCS.GetBody(CSharpSyntaxTree.ParseText(code).Single<LocalFunctionStatementSyntax>()).Should().BeNull();
+        }
+
+        [TestMethod]
+        public void GetBody_LocalFunctionStatement_BodyBlock()
+        {
+            var code = "class AClass { void AMethod() { int ALocalFunction() { var x1; var x2; var x3; return 42; } } }";
```

Comment:
This has compilation errors (var can not be inferred and maybe also unassigned values warnings).
```suggestion
            var code = "class AClass { void AMethod() { int ALocalFunction() { var x1 = 0; int x2 = 0; int x3 = 0; return 42; } } }";
```

Reply:
Ok for the compilation error being potentially confusing, although the test only parses and assess syntax.
I preferred shorter `int xN;` instructions, as they counted the number of statements in the body, that is assessed on the next line. I changed the code to use `i++` everywhere.

Reply:
`ALocalFunction` may still raise CS8321 The local function `ALocalFunction` is declared but never used. 
I would not complexify the code any further, with the sole goal of avoiding warnings.